File: C/stacks.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 10

struct Stack {
    int array[MAX_SIZE];
    int top;
};

void initStack(struct Stack* stack) {
    stack->top = -1;
}

int isFull(struct Stack* stack) {
    return stack->top == MAX_SIZE - 1;
}

int isEmpty(struct Stack* stack) {
    return stack->top == -1;
}

void push(struct Stack* stack, int data) {
    if (!isFull(stack)) {
        stack->array[++stack->top] = data;
        printf("Pushed: %d\n", data);
    } else {
        printf("Stack is full. Cannot push element.\n");
    }
}

void pop(struct Stack* stack) {
    if (!isEmpty(stack)) {
        printf("Popped: %d\n", stack->array[stack->top--]);
    } else {
        printf("Stack is empty. Cannot pop element.\n");
    }
}

int peek(struct Stack* stack) {
    if (!isEmpty(stack)) {
        return stack->array[stack->top];
    } else {
        printf("Stack is empty. Cannot peek.\n");
        return -1; // Assuming -1 indicates an error or an invalid value
    }
}

int main() {
    struct Stack myStack;
    initStack(&myStack);

    // Push elements onto the stack
    push(&myStack, 1);
    push(&myStack, 2);
    push(&myStack, 3);

    // Peek at the element on top of the stack
    printf("Top of the stack: %d\n", peek(&myStack));

    // Pop elements from the stack
    pop(&myStack);
    pop(&myStack);
    pop(&myStack);

    // Attempt to pop from an empty stack
    pop(&myStack);

    return 0;
}

File: Python/stacks.py
class Stack:
    def __init__(self):
        self.array = []
        self.top = -1

    def is_empty(self):
        return self.top == -1

    def is_full(self):
        return len(self.array) == 10

    def push(self, data):
        if not self.is_full():
            self.top += 1
            self.array.append(data)
            print(f"Pushed: {data}")
        else:
            print("Stack is full. Cannot push element.")

    def pop(self):
        if not self.is_empty():
            print(f"Popped: {self.array.pop()}")
            self.top -= 1
        else:
            print("Stack is empty. Cannot pop element.")

    def peek(self):
        if not self.is_empty():
            return self.array[self.top]
        else:
            print("Stack is empty. Cannot peek.")
            return -1


# Example usage
if __name__ == "__main__":
    my_stack = Stack()
    my_stack.push(1)
    my_stack.push(2)
    my_stack.push(3)
    print("Top of the stack:", my_stack.peek())
    my_stack.pop()
    my_stack.pop()
    my_stack.pop()
    my_stack.pop()
